### PR TITLE
Integration test

### PR DESCRIPTION
**Pull request by hshayesteh**
### Ready for review

### CODE REVIEW

File: server/gateway/api/swagger.yaml [hunk 1]
```diff
@@ -479,9 +489,16 @@ definitions:
         type: "string"
         enum: ["active", "inactive"]
         description: "The firstname of the user"
+      createdAt:
```
Comment (hadiosbourne) [inline server/gateway/api/swagger.yaml:492]: Does these have to be defined? 
If they are on the model they will be returned,
Are we modifying these in any ways?

File: server/gateway/testfile.js [hunk 2]
```diff
@@ -30,6 +29,7 @@ var excludedDir = ['endpoint'];
 // To run only specific unit tests add your class here in as
 // 'handlers/amqp/message-notification-handler-test.js'
 var customDir = [
+  'endpoint/user-test.js'
```
Comment (hadiosbourne) [inline server/gateway/testfile.js]: why hard code?
Won't it just read the whole test directory?

File: server/gateway/tests/endpoint/status-test.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+      .expect(200)
+      .end(function(err, res) {
+        if (err) return done(err);
+
+        validator.validate(res.body, schema).should.be.true;
+        done();
+      });
+    });
+
+    it('should respond with 405 Method not supported Error...', function(done) {
```
Comment (hadiosbourne) [inline server/gateway/tests/endpoint/status-test.js:85]: 405 tests are not necessary, swagger will handle these errors

File: server/user/index.js [hunk 4]
```diff
@@ -59,15 +59,45 @@ swaggerTools.initializeMiddleware(swaggerDoc, function callback(middleware) {
...
   subscriptionManager.initialize();
 
-  // Start the server
-  if (process.argv[2]) {
-    serverPort = process.argv[2];
+  if(process.argv.indexOf("PORT") != -1){ //does PORT exist?
+    serverPort = process.argv[process.argv.indexOf("PORT") + 1];
   }
 
+  // Start the server
```
Comment (hadiosbourne) [inline server/user/index.js]: commented code

**Review by hadiosbourne — CHANGES_REQUESTED** [review]

File: server/gateway/api/swagger.yaml (hunk 1 above)
Comment (hshayesteh) [reply server/gateway/api/swagger.yaml:492]: the swagger definitions are what is used to create the api documentation. If you omit properties then whoever client that consumes your api will not have the correct response structure. So you need to ensure the swagger documentation accurately reflect the actual response that is being returned

File: server/gateway/tests/endpoint/status-test.js (hunk 3 above)
Comment (hshayesteh) [reply server/gateway/tests/endpoint/status-test.js:85]: I know swagger is supposed to handle these errors but we should write the test to ensure swagger works as expected.

File: server/gateway/api/swagger.yaml (hunk 1 above)
Comment (hadiosbourne) [reply server/gateway/api/swagger.yaml:492]: 👍🏼

**Review by hadiosbourne — APPROVED** [review]

**Pull request merged**